### PR TITLE
[MRG] Patch writing iEEGCoordinateSystemDescription

### LINKED ISSUE #677: iEEGCoordinateSystemDescription is not being written correctly
#### Describe the bug
In my testing of #675 I was adding tests for `iEEGCoordinateSystemDescription` that we add to the `coordsystem.json` files. However, I noticed a bug that writes the description as `'n/a'`.

#### Steps to reproduce
In `test_write.py::test_coordsystem_json`, one can comment out the lines asserting what the CoordinateSystemDescriptions for iEEG should be and you will see that the tests fail.

#### Expected results
In order to overcome this, perhaps it's best to briefly summarize how we dealt with iEEG coordinate frames. There exist coordinate frames represented in three different spaces: i) MNE, ii) filename and iii) BIDS.

For example, in MNE, we'll represent "FreeSurfer" coordsystem as `'fs_tal'`. Since this isn't yet a supported coordinate system in BIDS, in BIDs, it would be represented as `'Other'`. In the filename, then this would be encoded as `space-fs` (note that we get rid of the underscore else it wouldn't be BIDS-compliant). Tbh this is quite confusing and unfortunate, but no way I see around it rn.

#### Possible Solutions
This is probably a mne-bidsv0.8+ problem, since it doesn't really have a huge impact. 

Possibly, a way for tackling this would be to refactor how `coordsystem.json` files are written, and keep track of a "MNE-coord-system" and "BIDS-coord-system" better, so they can be handled correctly in `dig.py::_coordsystem_json`.

#### Additional information
```
>>> sys_info()
Platform:      macOS-10.16-x86_64-i386-64bit
Python:        3.8.5 | packaged by conda-forge | (default, Jul 22 2020, 17:24:51)  [Clang 10.0.0 ]
Executable:    /Users/adam2392/opt/miniconda3/envs/mne/bin/python
CPU:           i386: 8 cores
Memory:        Unavailable (requires "psutil" package)
mne:           0.22.0
numpy:         1.19.1 {blas=NO_ATLAS_INFO, lapack=lapack}
scipy:         1.5.2
matplotlib:    3.3.0 {backend=TkAgg}

sklearn:       0.24.0
numba:         Not found
nibabel:       3.2.1
nilearn:       0.7.0
dipy:          Not found
cupy:          Not found
pandas:        1.1.3
mayavi:        Not found
pyvista:       0.26.1 {pyvistaqt=0.2.0, OpenGL 4.1 ATI-4.2.13 via AMD Radeon Pro 560 OpenGL Engine}
vtk:           9.0.1
PyQt5:         5.15.1
```

### PR DESCRIPTION
**Pull request by sappelhoff**
bad fix for #677 
closes #677 

But we need to refactor that part of the code eventually. cc @adam2392 

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [x] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

**Review by adam2392 — APPROVED** [review]
Sweet. Thanks for the fix!

File: mne_bids/dig.py [hunk 1]
```diff
@@ -269,11 +271,20 @@ def _coordsystem_json(*, raw, unit, hpi_coord_system, sensor_coord_system,
...
-    if sensor_coord_system == 'Other' and verbose:
-        print('Using the `Other` keyword for the CoordinateSystem field. '
-              'Please specify the CoordinateSystemDescription field manually.')
+    if sensor_coord_system == 'Other':
+        if verbose:
+            print('Using the `Other` keyword for the CoordinateSystem field. '
+                  'Please specify the CoordinateSystemDescription field '
+                  'manually.')
+        sensor_coord_system_descr = (COORD_FRAME_DESCRIPTIONS
+                                     .get(sensor_coord_system_mne.lower(),
```
Comment (adam2392) [inline mne_bids/dig.py:286]: It is possible that `sensor_coord_system_mne` is not available?

**Comment by adam2392** [comment]
When you say "bad fix" and this part needs to get refactored eventually, what do you think we should do?

**Comment by sappelhoff** [comment]
> When you say "bad fix" and this part needs to get refactored eventually, what do you think we should do?

I think the fix is good for now, but we should:

1. investigate this issue: https://github.com/mne-tools/mne-bids/blob/d65b8ca3edf38a4be5a1db8d24592cf2a1a3d5ef/mne_bids/tests/test_write.py#L2204-L2206

2. and this issue: https://github.com/mne-tools/mne-bids/issues/677#issuecomment-778302106

3. and start restricting what users can pass as valid value for the `space-*` entity (https://github.com/mne-tools/mne-bids/issues/677#issuecomment-778284872)

File: mne_bids/dig.py (hunk 1 above)
Comment (sappelhoff) [reply mne_bids/dig.py:286]: I don't know, but it's a good point --> I made this more robust in https://github.com/mne-tools/mne-bids/pull/706/commits/e734cefdb4815fc4033263e3ec4a55c941b6cf72

**Review by agramfort — APPROVED** [review]

File: mne_bids/dig.py [hunk 2]
```diff
@@ -239,8 +239,10 @@ def _coordsystem_json(*, raw, unit, hpi_coord_system, sensor_coord_system,
         as in BIDS_COORDINATE_UNITS.
     hpi_coord_system : str
         Name of the coordinate system for the head coils.
-    sensor_coord_system : str
+    sensor_coord_system : str | tuple of str
         Name of the coordinate system for the sensor positions.
+        If tuple of str, should be the BIDS coordinate frame, and
+        the MNE coordinate frame in that order.
```
Comment (hoechenberger) [inline mne_bids/dig.py:245]:
```suggestion
        If a tuple of strings, should be in the form:
        ``(BIDS coordinate frame, MNE coordinate frame)``.
```

File: mne_bids/dig.py [hunk 3]
```diff
@@ -269,11 +271,20 @@ def _coordsystem_json(*, raw, unit, hpi_coord_system, sensor_coord_system,
                          .format(coord_frame))
 
     # get the coordinate frame description
+    try:
+        sensor_coord_system, sensor_coord_system_mne = sensor_coord_system
+    except ValueError:  # pragma: no cover
```
Comment (hoechenberger) [inline mne_bids/dig.py:276]: why `no cover`?

File: mne_bids/dig.py [hunk 4]
```diff
@@ -269,11 +271,20 @@ def _coordsystem_json(*, raw, unit, hpi_coord_system, sensor_coord_system,
...
     sensor_coord_system_descr = (COORD_FRAME_DESCRIPTIONS
                                  .get(sensor_coord_system.lower(), "n/a"))
-    if sensor_coord_system == 'Other' and verbose:
-        print('Using the `Other` keyword for the CoordinateSystem field. '
-              'Please specify the CoordinateSystemDescription field manually.')
+    if sensor_coord_system == 'Other':
+        if verbose:
+            print('Using the `Other` keyword for the CoordinateSystem field. '
+                  'Please specify the CoordinateSystemDescription field '
+                  'manually.')
```
Comment (hoechenberger) [inline mne_bids/dig.py:284]: This should use the logger

**Review by hoechenberger — CHANGES_REQUESTED** [review]

File: mne_bids/dig.py (hunk 3 above)
Comment (sappelhoff) [reply mne_bids/dig.py:276]: initially I thought this is never reached in our code base, but I was wrong. I'll remove this pragma

**Review by hoechenberger — APPROVED** [review]

**Comment by hoechenberger** [comment]
Thanks @sappelhoff

**Pull request merged**